smartcontract/service/native/global_params/global_params.go

```diff
@@ -220,22 +249,37 @@ func CreateSnapshot(native *native.NativeService) ([]byte, error) {
 	return utils.BYTE_TRUE, nil
 }
 
-func getAdmin(native *native.NativeService, contract common.Address) {
-	if admin == nil || *admin == *new(Admin) {
-		var err error
-		// get admin from database
-		admin, err = GetStorageAdmin(native, GenerateAdminKey(contract, false))
-		// there are no admin in database
+func getAdmin(native *native.NativeService, contract common.Address) *Role {
+	// get admin from database
+	admin, err := GetStorageRole(native, generateAdminKey(contract, false))
+	// there are no admin in database
+	if err != nil {
+		bookKeeepers, err := config.DefConfig.GetBookkeepers()
+		if err != nil {
+			log.Errorf("GetBookkeepers error: %v", err)
+			return nil
+		}
+		initAddress := ctypes.AddressFromPubKey(bookKeeepers[0])
+		copy((*admin)[:], initAddress[:])
+	}
+	return admin
+}
+
+func getOperator(native *native.NativeService, contract common.Address) *Role {
+	var err error
+	// get operator from database
+	operator, err := GetStorageRole(native, GenerateOperatorKey(contract))
+	// there are no operator in database
+	if err != nil {
+		bookKeeepers, err := config.DefConfig.GetBookkeepers()
```

Comment:
question:
If there's no operator, take Admin as default?

Reply:
it has set operator to admin in contract init function